core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -3913,8 +3915,8 @@ private RecordCursorFactory generateSampleBy(
                         false,
                         groupByFunctions,
                         groupByFunctionPositions,
-                        outerProjectionFunctions,
-                        innerProjectionFunctions,
+                        tempOuterProjectionFunctions,
```

Comment:
nit: it would be good to clear these collections on exit from `generateSampleBy()` method